File: server/go/upsert-user.go
//go:build ignore

// +build ignore

package main

import (
	"context"
	"fmt"
	"os"
	"path/filepath"

	"github.com/joho/godotenv"
	"github.com/trycourier/courier-go/v4"
	"github.com/trycourier/courier-go/v4/option"
)

func LoadEnv() {
	envPath := filepath.Join("..", ".env")
	godotenv.Load(envPath)
}

func main() {
	LoadEnv()

	apiKey := os.Getenv("COURIER_API_KEY")
	userID := os.Getenv("COURIER_UPSERT_USER_USER_ID")
	email := os.Getenv("COURIER_UPSERT_USER_EMAIL")
	name := os.Getenv("COURIER_UPSERT_USER_NAME")
	phoneNumber := os.Getenv("COURIER_UPSERT_USER_PHONE_NUMBER")

	client := courier.NewClient(
		option.WithAPIKey(apiKey),
	)

	profile := make(map[string]interface{})
	if email != "" { profile["email"] = email }
	if name != "" { profile["name"] = name }
	if phoneNumber != "" { profile["phone_number"] = phoneNumber }

	requestBody := map[string]interface{}{
		"profile": profile,
	}

	var response map[string]interface{}
	err := client.Post(
		context.Background(),
		fmt.Sprintf("/profiles/%s", userID),
		requestBody,
		&response,
	)

	if err != nil {
		panic(err.Error())
	}

	fmt.Printf("%+v\n", response)
}

File: server/ruby/upsert-user.rb
require 'dotenv'
require 'trycourier'

env_path = File.join(File.dirname(__FILE__), '..', '.env')
Dotenv.load(env_path)

api_key = ENV['COURIER_API_KEY']
user_id = ENV['COURIER_UPSERT_USER_USER_ID']
email = ENV['COURIER_UPSERT_USER_EMAIL']
name = ENV['COURIER_UPSERT_USER_NAME']
phone_number = ENV['COURIER_UPSERT_USER_PHONE_NUMBER']

client = Trycourier::Client.new(api_key: api_key)

profile = {}
profile[:email] = email if email && !email.empty?
profile[:name] = name if name && !name.empty?
profile[:phone_number] = phone_number if phone_number && !phone_number.empty?

response = client.profiles.create(user_id, profile: profile)

puts response
